core/sail/base/src/main/java/org/eclipse/rdf4j/sail/base/SailSourceConnection.java

```diff
@@ -135,10 +141,17 @@
 	 * @param resolver
 	 */
 	protected SailSourceConnection(AbstractSail sail, SailStore store, FederatedServiceResolver resolver) {
+		this(sail, store, new SimpleEvaluationStrategyFactory(resolver), resolver);
```

Comment:
I see what you're doing. If we go this route, it needs to be clear at the interface level what is expected. We possible need to add a setter-method to either Sail, or SailConnection, to indicate that EvaluationStrategyFactory is something that can be injected from outside.